utility/block.go

```diff
@@ -21,25 +24,26 @@ import (
 */
 
 // TODO: Make sure to call `utility.CheckTransaction`, which calls `persistence.TransactionExists`
-func (u *UtilityContext) CreateAndApplyProposalBlock(proposer []byte, maxTransactionBytes int) (stateHash string, transactions [][]byte, err error) {
-	lastBlockByzantineVals, err := u.GetLastBlockByzantineValidators()
+func (u *utilityContext) CreateAndApplyProposalBlock(proposer []byte, maxTransactionBytes int) (stateHash string, transactions [][]byte, err error) {
+	lastBlockByzantineVals, err := u.getLastBlockByzantineValidators()
 	if err != nil {
 		return "", nil, err
 	}
 	// begin block lifecycle phase
-	if err := u.BeginBlock(lastBlockByzantineVals); err != nil {
+	if err := u.beginBlock(lastBlockByzantineVals); err != nil {
 		return "", nil, err
 	}
 	transactions = make([][]byte, 0)
 	totalTxsSizeInBytes := 0
 	txIndex := 0
-	mempool := u.GetBus().GetUtilityModule().GetMempool()
+
+	mempool := u.getBus().GetUtilityModule().GetMempool()
 	for !mempool.IsEmpty() {
 		txBytes, err := mempool.PopTx()
 		if err != nil {
 			return "", nil, err
 		}
-		transaction, err := typesUtil.TransactionFromBytes(txBytes)
+		transaction, err := typesUtil.TxFromBytes(txBytes)
```

Comment:
Not a strong opinion: shouldn't this be `tx` now? Actually I'd consider renaming the type as well from `Transaction` to `Tx`. I know it's interchangeable but since we are touching it, let's give it a little bit more love?